src/timepicker/TimePicker.ts

```diff
@@ -0,0 +1,481 @@
+	minute?: number;
+	second?: number;
+}
+
+/**
+ * @type TimePickerProperties
+ *
+ * Properties that can be set on a TimePicker component
+ *
+ * @property autoBlur           Determines whether the input should blur after value selection
+ * @property buttonText         The text for the menu trigger (defaults to '')
+ * @property customOptionItem   Can be used to render a custom option
+ * @property customOptionMenu   Can be used to render a custom option menu
+ * @property disabled           Prevents user interaction and styles content accordingly
+ * @property end                The maximum time to display in the menu (defaults to '23:59:59')
+ * @property formId             ID of a form element associated with the form field
+ * @property getOptionLabel     Can be used to get the text label of an option based on the underlying option object
+ * @property inputProperties    TextInput properties to set on the underlying input
+ * @property isOptionDisabled   Used to determine if an item should be disabled
+ * @property invalid            Determines if this input is valid
+ * @property label              Label settings for form label text, position, and visibility
+ * @property onBlur             Called when the input is blurred
+ * @property onChange           Called when the value changes
+ * @property onFocus            Called when the input is focused
+ * @property onRequestOptions   Called when options are shown; should be used to set `options`
+ * @property openOnFocus        Determines whether the result list should open when the input is focused
+ * @property options            Options for the current input; should be set in response to `onRequestOptions`
+ * @property readOnly           Prevents user interaction
+ * @property required           Determines if this input is required, styles accordingly
+ * @property start              The minimum time to display in the menu (defaults to '00:00:00')
+ * @property step               The number of seconds between each option in the menu (defaults to 60)
+ * @property value              Value to set on the input
+ */
+export interface TimePickerProperties extends ThemeableProperties {
+	autoBlur?: boolean;
+	buttonText?: string;
+	customOptionItem?: any;
+	customOptionMenu?: any;
+	disabled?: boolean; // defaults to false
+	end?: string | TimeUnits;
+	formId?: string;
+	getOptionLabel?: (option: TimeUnits) => string;
+	inputProperties?: TextInputProperties;
+	invalid?: boolean; // defaults to false
+	isOptionDisabled?(option: TimeUnits): boolean;
+	label?: string | LabelOptions;
+	onBlur?: (value: string) => void;
+	onChange?: (value: string) => void;
+	onFocus?: (value: string) => void;
+	onRequestOptions?(value: string, options: ReadonlyArray<TimeUnits>): void;
+	openOnFocus?: boolean;
+	options?: TimeUnits[];
+	readOnly?: boolean;  // defaults to false
+	required?: boolean; // defaults to false
+	start?: string | TimeUnits;
+	step?: number; // defaults to 60
+	value?: string;
```

Comment:
The issue excludes the fallback to native from the initial implementation. If @rishson agrees to hold off landing until that is added, I'll do so.